### PR TITLE
Fixes #29729 - add force unlock to tasks table rows

### PR DESCRIPTION
**Pull request by MariaAga**
Each task - row should have a force cancel button (if the task is not already stopped)
Does not include unlock for multiple tasks
depends on https://github.com/theforeman/foreman-tasks/pull/539

### CODE REVIEW

**Comment by adamruzicka** [comment]
Needs a rebase

**Comment by adamruzicka** [comment]
Needs a rebase

File: webpack/ForemanTasks/Components/TaskActions/UnlockModals.js [hunk 1]
```diff
@@ -22,12 +22,12 @@ export const UnlockModal = ({ onClick, id }) => (
...
 );
 
-export const ForceUnlockModal = ({ onClick, id }) => (
+export const ForceUnlockModal = ({ onClick, id, selectedRowsLen }) => (
   <ClickConfirmation
     id={id}
     title={__('Force Unlock')}
     body={__(
-      'Resources will be unlocked and will not prevent other tasks from being run. As the task might be still running, it should be avoided to use this unless you are really sure the task got stuck'
+      `Resources for ${selectedRowsLen} task(s) will be unlocked and will not prevent other tasks from being run. As the task might be still running, it should be avoided to use this unless you are really sure the task got stuck.`
```
Comment (sharvit) [inline webpack/ForemanTasks/Components/TaskActions/UnlockModals.js:30]: I don't think the translation work that way, see: http://theforeman.github.io/foreman/?path=/docs/introduction-internationalization--page

File: webpack/ForemanTasks/Components/TasksTable/Components/ConfirmModal/ConfirmModal.js [hunk 2]
```diff
@@ -16,32 +18,46 @@ export const ConfirmModal = ({
...
-      ),
-      { action: actionText, number: selectedRowsLen, state: actionState }
-    )}
-    <ForemanModal.Footer>
-      <Button onClick={setModalClosed}>{__('No')}</Button>
-      <Button
-        className="confirm-button"
-        bsStyle="primary"
+}) => {
+  if (actionType === FORCE_UNLOCK_MODAL) {
```
Comment (sharvit) [inline webpack/ForemanTasks/Components/TasksTable/Components/ConfirmModal/ConfirmModal.js:22]: It doesn't feel right, why not using the `ForceUnlockModal` from the parent instead?

Comment (MariaAga) [reply webpack/ForemanTasks/Components/TasksTable/Components/ConfirmModal/ConfirmModal.js:22]: Can you explain what do you mean to use it from the parent?

File: webpack/ForemanTasks/Components/TaskActions/UnlockModals.js (hunk 1 above)
Comment (MariaAga) [reply webpack/ForemanTasks/Components/TaskActions/UnlockModals.js]: You're right, thanks!

**Comment by MariaAga** [comment]
Which part is confusing? Maybe I can add/remove something to make it less confusing

File: webpack/ForemanTasks/Components/TasksTable/Components/ConfirmModal/ConfirmModal.js (hunk 2 above)
Comment (sharvit) [reply webpack/ForemanTasks/Components/TasksTable/Components/ConfirmModal/ConfirmModal.js:22]: I guess there is a parent component that rendering `<ConfirmModal ... />` to render `ForemanModal` and `<ConfirmModal ... actionType={FORCE_UNLOCK_MODAL} />` to render `ForceUnlockModal`.

Why not just rendering `ForceUnlockModal` instead `<ConfirmModal ... actionType={FORCE_UNLOCK_MODAL} />`?

**Comment by sharvit** [comment]
> @sharvit Which part is confusing? Maybe I can add/remove something to make it less confusing

I see it confusing having switch inside another switch but It's not part of this PR right?

**Comment by MariaAga** [comment]
@sharvit It was made to split the logic so it won't all be in the `TasksTablePage`, it was part of the previous PR which was merged: https://github.com/theforeman/foreman-tasks/pull/538

File: webpack/ForemanTasks/Components/TasksTable/Components/ConfirmModal/ConfirmModal.js (hunk 2 above)
Comment (MariaAga) [reply webpack/ForemanTasks/Components/TasksTable/Components/ConfirmModal/ConfirmModal.js:22]: The parent component doesn't have the info about actionType

**Comment by sharvit** [comment]
> @sharvit It was made to split the logic so it won't all be in the `TasksTablePage`, it was part of the previous PR which was merged: #538

I see, would it make sense just moving the inner switch to a separate function to increase readability?

**Comment by MariaAga** [comment]
@sharvit  I don't think it would make sense as the whole reducer is just those 2 switches with nothing else

**Comment by adamruzicka** [comment]
Thank you @MariaAga & @sharvit !

**Pull request merged**